examples/todo-list/src/__tests__/acceptance/todo-list-image.acceptance.ts

```diff
@@ -55,7 +55,8 @@ describe('TodoListApplication', () => {
     expect(response.body).to.containEql(expected);
 
     const created = await todoListImageRepo.findById(response.body.id);
-    expect(toJSON(created)).to.deepEqual({id: response.body.id, ...expected});
+    expected.id = response.body.id;
+    expect(toJSON(created)).to.deepEqual(expected);
```

Comment:
Can you please explain what error is TypeScript 3.9 reporting here? I'd like to better understand what's going on and preferably find a way how to avoid mutation of `expected` object.

Reply:
TypeScript reports an error for `expect(toJSON(created)).to.deepEqual({id: response.body.id, ...expected});` with the following reasoning:

1. The inferred type of `expected` has an `id` (non-optional) property.
2. The spread assignment has overriding property `id` from the 1st and 2nd argument. TypeScript think the 1st `id` will be ALWAYS overridden and it is probably a bug in the code.
3. In our case, `expected.id` is not defined when it's instantiated from a model class without `create`.